BUILDING.md

```diff
@@ -129,22 +130,50 @@ $ make
 
 If you want to build shared libraries (`.so` files), run `cmake` with `-DBUILD_SHARED_LIBS=ON`.
 
-### Windows, Using Visual Studio 2015 or 2017
+
+### Windows, Using CMake
 
 When using the "Visual Studio" generator,
 cmake will generate a solution (`grpc.sln`) that contains a VS project for
-every target defined in `CMakeLists.txt` (+ a few extra convenience projects
+every target defined in `CMakeLists.txt` (plus a few extra convenience projects
 added automatically by cmake). After opening the solution with Visual Studio
 you will be able to browse and build the code.
+
+> **Note**
+> A Visual C++ compiler is needed when building using CMake directly, which you specify as a [cmake-generator](https://cmake.org/cmake/help/latest/manual/cmake-generators.7.html#manual:cmake-generators(7)) option.
+
 ```
 > @rem Run from grpc directory after cloning the repo with --recursive or updating submodules.
 > md .build
 > cd .build
 > cmake .. -G "Visual Studio 14 2015"
 > cmake --build . --config Release
 ```
+> **Note** If you need to figure our our Visual Studio version, Wikipedia details the [mapping between Visual Studio product name and version number](https://en.wikipedia.org/wiki/Microsoft_Visual_Studio) in the 'History' section (e.g. Visual Studio 2022 == Version 17.0).
+
+### Windows, Using Visual Studio
```

Comment:
I don't see the value of this section. Since you need to run cmake commands anyway to generate the .sln file anyway, the commandline instructions are basically identical to the previous section and building in Visual Studio once you have a .sln file is very straightforward.

I think it would be enough if in the previous section you just added a note that the result of cmake configure step can be opened and build in visual studio - no need to add a new section and multiple paragraphs for this.

Reply:
Yeah, the code block describing the steps at the cmd-line are quite similar so that can be shorted to just highlight the difference. I copied the code block for consistency but happy to modify.

The reasons why I put it in its own section are:

- The **Windows, Using CMake** section is really for command-line enthusiasts and many VC++ developers will have a natural preference to build inside Visual Studio. 
- I figured there really should be instructions for a Windows build that cross-referenced the [gRPC Quick start](https://grpc.io/docs/languages/cpp/quickstart) page. VC++ developers are going to find their way to this page because it doesn't include instructions for Windows.
- It was more in keeping with the existing layout with there being a separate section covering  **Windows, Using Ninja**.

Ultimately, it seems you can build on Windows using CMake, Visual Studio or using Ninja. All three use CMake.

Reply:
As mentioned in a related comment, I've revamped this section to better illustrate the alternative approach. Could I please ask you to review again (in context of the general flow of the doc)? Thanks.